File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {

        if nums.len() as i32 <= 1 {
            return nums.len() as i32;
        }

        let mut index : usize = 0;

        for i in 1..nums.len() {
            if nums[index] != nums[i] {
                index += 1;
                nums[index] = nums[i];
                continue;
            }
        }

        index += 1;
        nums.truncate(index);

        index as i32
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> set = new HashSet<>();

        int keeper = 0;
        for (int i = 0; i < nums.length; i++) {

            if (set.contains(nums[i])) {
                continue;
            }

            nums[keeper] = nums[i];
            set.add(nums[i]);
            keeper++;
        }

        return keeper;
    }
}